app/controllers/users/contact_details.js

```diff
@@ -0,0 +1,165 @@
+import Ember from "ember";
+import AjaxPromise from "stock/utils/ajax-promise";
+import AsyncMixin, { ERROR_STRATEGIES } from "stock/mixins/async";
+import TitleAndLanguageMixin from "stock/mixins/grades_option";
+import ImageUploadMixin from "stock/mixins/image_upload";
+
+export default Ember.Controller.extend(
+  AsyncMixin,
+  ImageUploadMixin,
+  TitleAndLanguageMixin,
+  {
+    user: Ember.computed.alias("model.user"),
+    userService: Ember.inject.service(),
+
+    emailIdNotPresent: Ember.computed("user.email", function() {
+      let email = this.get("user.email");
+      return !/^[^@\s]+@[^@\s]+/.test(email);
+    }),
+
+    mobileNotPresent: Ember.computed(
+      "user.mobileWithoutCountryCode",
+      function() {
+        let mobile = this.get("user.mobileWithoutCountryCode");
+        return !/^[456789]\d{7}/.test(mobile);
+      }
+    ),
+
+    districts: Ember.computed(function() {
+      return this.get("store")
+        .peekAll("district")
+        .sortBy("name");
+    }),
+
+    getUpdatedDistrictParams() {
+      let address_attributes = {
+        district_id: this.get("user.associatedDistrict.id") || null
+      };
+      return { address_attributes };
+    },
+
+    saveImage() {
+      let image = this.get("newUploadedImage");
```

Comment:
this method should be designed as the name suggests saveImage, meaning ONLY to save the image.

```
saveImage(img){
  image.save()
}
```

I would suggest to make the image creation/deletion as a service as I am seeing ur using it in multiple places

```javascript
export default ApiBaseService.extend({
   create(img) {
      // logic
  },
 
 delete(img) {
    // logic
 }
})

```

Reply:
sure!! seems correct to have it in service